src/main/io/debugserial.c

```diff
@@ -0,0 +1,125 @@
+
+#include "platform.h"
+
+#ifdef USE_DEBUG_SERIAL
+
+#include "io/serial.h"
+#include "common/printf.h"
+#include "io/debugserial.h"
+
+#define INDENTATION_CHAR " "
+#define INDENTATION_LENGTH 3
+#define CLEAR_SEQUENCE "%s", "\x1b[2J\x0"
+#define INITIALIZATION_MESSAGE "DebugSerial_Initialize: OK\r\n"
+#define CHECK_ENABLED if(!debugSerial.enabled) return
```

Comment:
Hiding flow control in macro is usually very bad idea, it will confuse anyone reading or debugging code.
Also function-like macro should behave like single statement - wrapping it in `do { <code> } while(0)` is common way to enforce this (naked if will interfere with `else` - `if(test) CHECK_ENABLED; else statement();`)